metagov/metagov/core/views.py

```diff
@@ -35,150 +42,16 @@ def index(request):
-    """
-    cls = plugin_registry[plugin_name]
-
-    @community_middleware
-    @api_view(["POST"])
-    def enable_plugin(request):
-        plugin_config = JSONParser().parse(request)
-        # Create or re-create the plugin (only one instance per community supported for now!)
-        plugin, created = utils.create_or_update_plugin(plugin_name, plugin_config, request.community)
-        # Serialize and return the Plugin instance
-        serializer = PluginSerializer(plugin)
-        resp_status = status.HTTP_201_CREATED if created else status.HTTP_200_OK
-        return JsonResponse(serializer.data, status=resp_status)
-
-    request_body_schema = utils.json_schema_to_openapi_object(cls.config_schema) if cls.config_schema else {}
-
-    return swagger_auto_schema(
-        method="post",
-        responses={
-            201: openapi.Response(
-                "Plugin enabled",
-                PluginSerializer,
-            ),
-            200: openapi.Response(
-                "The Plugin was already enabled. Plugin was updated if the config changed.",
-                PluginSerializer,
-            ),
-        },
-        operation_id=f"Enable {plugin_name}",
-        tags=[Tags.COMMUNITY],
-        operation_description=f"Enable {plugin_name} plugin.",
-        manual_parameters=[MetagovSchemas.community_header],
-        request_body=openapi.Schema(
-            type=openapi.TYPE_OBJECT,
-            properties={
-                **request_body_schema.get("properties", {}),
-            },
-            required=request_body_schema.get("required", []),
-        ),
-    )(enable_plugin)
-
-
-@swagger_auto_schema(
-    method="delete",
-    operation_id="Disable plugin",
-    operation_description="Delete a plugin instance. This is an irreversible action.",
-    manual_parameters=[MetagovSchemas.plugin_name_in_path],
-    responses={204: "Plugin disabled successfully"},
-    tags=[Tags.COMMUNITY],
-)
-@api_view(["DELETE"])
-def delete_plugin(request, plugin_name, id):
-    try:
-        plugin = Plugin.objects.get(pk=id)
-    except Plugin.DoesNotExist:
-        return HttpResponseNotFound()
-    logger.info(f"Deleting plugin {plugin}")
-    plugin.delete()
-    return HttpResponse(status=status.HTTP_204_NO_CONTENT)
-
-
-@swagger_auto_schema(**MetagovSchemas.plugin_authorize)
-@api_view(["GET"])
-def plugin_authorize(request, plugin_name):
-    return metagov_handler.handle_oauth_authorize(request, plugin_name)
-
-
 @swagger_auto_schema(method="GET", auto_schema=None)
 @api_view(["GET"])
 def plugin_auth_callback(request, plugin_name):
+    """This function provides endpoints for plugins to provide to external platforms so that platforms
+    can redirect ("call back") after oauth is complete."""
     return metagov_handler.handle_oauth_callback(request, plugin_name)
 
 
+# FIXME: it feels like this should be in HTTPWrapper but might a Django-based driver want a
+# convenience method for accessing schemas?
```

Comment:
Yes- I made this accessible to Django-based drivers  on the MetagovApp class on my branch. You can leave it for now and I'll put in the change. The branch is still a mess right now, but here's a link to the specific diff https://github.com/metagov/metagov-prototype/compare/try-import#diff-bea35d1bc007624e5ca7a375526873b6dc9c1d041ddc1cdd691fe7f0c4ce9e8dR20

Reply:
Gotcha, I left it in for now. Once it's handled we can get rid of core/views.py and make the root-level urls.py file truly minimal.